hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/protocol/SecretKeyProtocolScm.java

```diff
@@ -28,4 +31,14 @@
     clientPrincipal = HDDS_SCM_KERBEROS_PRINCIPAL_KEY
 )
 public interface SecretKeyProtocolScm extends SecretKeyProtocol {
+
+  /**
+   * Force generates new secret keys (rotate).
+   *
+   * @param force boolean flag that forcefully rotates the key on demand
+   * @return key rotation status
+   * @throws TimeoutException
+   * @throws IOException
+   */
+  boolean checkAndRotate(boolean force) throws TimeoutException, IOException;
```

Comment:
As below comments, `TimeoutException` should not be exposed to client side.

Reply:
Understood, thanks @duongkame, made the required changes!